ios/sdk/component/image/HippyImageView.m

```diff
@@ -265,6 +268,20 @@ - (void)setRenderingMode:(UIImageRenderingMode)renderingMode
 	}
 }
 
+- (BOOL)shouldChangeImageSource
+{
+// We need to reload if the desired image source is different from the current image
+// source AND the image load that's pending
+    NSDictionary *source = [self.source firstObject];
+    if (source) {
+        NSString *desiredImageSource = source[@"uri"];
+  
+        return ![desiredImageSource isEqual:self.imageSourceUri] &&
```

Comment:
[suggest] use 'isEqualToString:' method instead of 'isEqual:' if we know source[@"uri"] is NSString type.